docs/plugins/data_persistence.md

```diff
@@ -0,0 +1,232 @@
+# Data persistence
+There are four plugins that currently have data persistence ability in opensearch dashboard: dashboard, discover, timeline, visualize, and vis-builder. They are using the following services and mechanisms from opensearch_dashboard_utils plugin to achieve data persistence.
```

Comment:
Could we add a definition of Data persistence?

Reply:
Added some explanations for data persistence.